driver-core/src/main/com/mongodb/operation/BsonArrayWrapper.java

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright 2015 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.operation;
+
+import org.bson.BsonArray;
+import org.bson.BsonInvalidOperationException;
+import org.bson.BsonValue;
+import org.bson.BsonWriter;
+import org.bson.codecs.Encoder;
+import org.bson.codecs.EncoderContext;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ListIterator;
+
+import static org.bson.assertions.Assertions.notNull;
+
+
+class BsonArrayWrapper<T> extends BsonArray {
```

Comment:
Most of this class appears to be unused and untested.  In particular, getUnwrapped is never called.